hecat/processors/github_metadata.py

```diff
@@ -76,41 +67,197 @@ def write_software_yaml(step, software):
     with open(dest_file, 'w+', encoding="utf-8") as yaml_file:
         yaml.dump(software, yaml_file)
 
+def extract_repo_name(url):
+    re_result = re.search(r'^https?:\/\/github\.com\/[^\/]+\/([^\/]+)\/?$', url)
+    return re_result.group(1) if re_result else None
+
 def add_github_metadata(step):
     """gather github project data and add it to source YAML files"""
     GITHUB_TOKEN = os.environ['GITHUB_TOKEN']
-    g = github.Github(GITHUB_TOKEN)
     errors = []
+    github_projects = []
+    # Load software data
     software_list = load_yaml_data(step['module_options']['source_directory'] + '/software')
     logging.info('updating software data from Github API')
+    # Check if the source code URL is a GitHub repository and add it to the queue to be processed
     for software in software_list:
-        github_url = ''
         if 'source_code_url' in software:
             if re.search(r'^https://github.com/[\w\.\-]+/[\w\.\-]+/?$', software['source_code_url']):
-                github_url = software['source_code_url']
+                # Check if we only want to update missing metadata or all metadata
+                if 'gh_metadata_only_missing' in step['module_options'].keys() and step['module_options']['gh_metadata_only_missing']:
+                    if ('stargazers_count' not in software) or ('updated_at' not in software) or ('archived' not in software) or ('current_release' not in software) or ('last_commit_date' not in software) or ('commit_history' not in software):
+                        github_projects.append(software)
+                    else:
+                        logging.debug('all metadata already present, skipping %s', software['source_code_url'])
+                # If key is not present, update all metadata
+                else:
+                    github_projects.append(software)
+        # TODO: Why do we need to check the website_url? We can exspect that the source_code_url is always present and the website_url is optional and even if changed it would not point to a github repository
```

Comment:
You are correct. I think the entire `elif 'website_url' in software:` block can be removed